src/picosvg/geometric_types.py

```diff
@@ -44,13 +44,13 @@ def __sub__(self, other: _PointOrVec) -> _PointOrVec:
             return self._sub_pt(other)
         elif isinstance(other, Vector):
             return self._sub_vec(other)
-        return NotImplemented
+        raise ValueError(f"Other must be a Point or Vector, not a {type(other)}")
 
     def __add__(self, other: "Vector") -> "Point":
         """Return Point translated by other Vector"""
         if isinstance(other, Vector):
             return self.__class__(self.x + other.x, self.y + other.y)
-        return NotImplemented
+        raise ValueError(f"Other must be a Vector, not a {type(other)}")
```

Comment:
Changes here are due to typechecker complaining; I don't want the return type to be Point or NotImplemented

Reply:
but the canonical way to implement math operator overrides is exactly to return NotImplemented when the other type isn't among the ones which the current method is capable of handling, this enables the interpreter to check if the same operator is implemented by the other side of the expression, see https://docs.python.org/3/library/numbers.html#implementing-the-arithmetic-operations

I'm surprised that pytype doesn't automatically support this..

Reply:
there's even an explicit test for this in pytype test suite:

https://github.com/google/pytype/blob/b6b6448dc562a7200326c92e75efeed203984e16/pytype/tests/test_cmp2.py#L53-L78

Reply:
Huh. TIL. I will revert this part.